src/Transactions/AccountSignatureMap.cs

```diff
@@ -56,4 +56,66 @@ public Grpc.V2.AccountSignatureMap ToProto()
         }
         return accountSignatureMap;
     }
+
+    internal static AccountSignatureMap From(Grpc.V2.AccountSignatureMap map)
+    {
+        var dict = new Dictionary<AccountKeyIndex, byte[]>();
+        foreach (var s in map.Signatures)
+        {
+            dict.Add(new AccountKeyIndex((byte)s.Key), s.Value.Value.ToByteArray());
+        }
+        return Create(dict);
+    }
+}
+
+
+/// <summary>Index of a key in an authorizations update payload.</summary>
+public sealed record UpdateKeysIndex(byte Value);
+
+/// <summary>A map from <see cref="UpdateKeysIndex"/> to signatures.</summary>
+public sealed record UpdateInstructionSignatureMap
```

Comment:
I am not sure, I understand the issue here, it is already a `class` (`record class`), and to my knowledge this is only changing the behavior of equality.
Also, I don't see this type being mutated and is only used for reading from the chain, meaning immutability is not an issue.

Similarly, for the user will encounter `AccountSignatureMap` when reading from chain, and probably only be constructed internally in the SDK and not by the user.

Reply:
Having read a bit more about records, I think they are the right thing here, since the type is just holding data and does not provide any (interesting) logic. The generated `toString` function is useful as well. I suggest that we fix the behavior of equality instead.

Changing this would also make it inconsistent with the nearly identical type `AccountSignatureMap`.